### PR TITLE
Toggle note tag names

### PR DESCRIPTION
**Pull request by cutandpastey**
This PR adds the ability to click a note and toggle it's tag name i.e. `gu-flag` > `gu-correct`. This has to be configured during plugin init.

### QA
- Create a flag (F9), then click the give flag.  The tag name should have changed to `gu-correct`.

### CODE REVIEW

File: src/actions/noting/reset-note-segment-classes.js [hunk 1]
```diff
@@ -29,6 +37,6 @@ module.exports = function updateStartAndEndClasses(noteSegments) {
   addClass(noteSegments[0].vNode, 'note--start');
   addClass(noteSegments[noteSegments.length - 1].vNode, 'note--end');
 
-  return noteSegments;
+  return noteSegments
```
Comment (robinedman) [inline src/actions/noting/reset-note-segment-classes.js]: Missing a semi-colon?

File: src/generate-note-controller.js [hunk 2]
```diff
@@ -41,7 +42,7 @@ module.exports = function(scribe, attrs){
 
       //scribe command events
       emitter.on('command:note', tag => this.note(tag));
-      emitter.on('command:toggle:single-note', tag => this.toggleSelectedNotes(tag));
+      emitter.on('command:toggle:single-note', tag => this.toggleSelectedNotesCollapseState(tag));
       emitter.on('command:toggle:all-notes', tag => this.toggleAllNotes(tag));
```
Comment (robinedman) [inline src/generate-note-controller.js]: Maybe call this `toggleAllNotesCollapseState` for consistency

File: src/generate-note-controller.js [hunk 3]
```diff
@@ -75,24 +76,88 @@ module.exports = function(scribe, attrs){
...
-          range.selectNodeContents(e.target);
-          vSelection.selection.removeAllRanges();
-          vSelection.selection.addRange(range);
-          this.toggleSelectedNotes();
+        this.toggleClickedNotesTagNames(e.target);
+        break;
+
+        default:
+          e.preventDefault();
+        this.toggleClickedNotesCollapseState(e.target);
```
Comment (robinedman) [inline src/generate-note-controller.js]: Indentation

**Comment by robinedman** [comment]
The code looks really good to me!

The problem I see is more of a usability thing (@alastairjardine?)  : How do you "unnote" a _correct_? The only way I can see is to turn the _correct_ into a _flag_ (by clicking at the start/end or pressing F9) and then pressing F9 again.

**Comment by alastairjardine** [comment]
Cycling through seems sensible. F9 to add a flag, ⇧F9 to add a correct directly, F9 on correct to toggle to nothing?

**Comment by robinedman** [comment]
That sounds good I think. Easy to implement @cutandpastey ?

**Comment by alastairjardine** [comment]
It would have to maintain the selection so as not to be too annoying if the person removes the correct in correctly.

Or…⌥F9 to remove?

**Comment by cutandpastey** [comment]
You can only note/un-note something with the same key as you create one.  So for example a flag is F9 to create or remove.  I can add ⇧F9  to create or remove flag if you want?

File: example/example.html [hunk 4]
```diff
@@ -50,10 +50,11 @@
...
             user: "Example User",
             scribeInstancesSelector: '.scribe',
-            selectors: [
-              { commandName: 'note', tagName: 'gu-note', keyCodes: [119, 121, {'altKey': 8}] },
-              { commandName: 'flag', tagName: 'gu-flag', keyCodes: [120] }
-            ]
+              selectors: [
+                { commandName: 'note',    tagName: 'gu-note',    clickAction: 'collapse',   keyCodes: [119, 121, {'altKey': 8}] },
+                { commandName: 'flag',    tagName: 'gu-flag',    clickAction: 'toggle-tag', toggleTagTo: 'gu-correct', keyCodes: [120] },
+                { commandName: 'correct', tagName: 'gu-correct', clickAction: 'toggle-tag', toggleTagTo: 'gu-flag', keyCodes: [] }
```
Comment (hmgibson23) [inline example/example.html]: Are these the right way round?

**Comment by hmgibson23** [comment]
:+1:

**Comment by cutandpastey** [comment]
Closing due to https://github.com/guardian/scribe-plugin-noting/pull/95

**Pull request closed without merging**